### PR TITLE
Update workflow to validate all policies according to the contribution guide 

### PR DESCRIPTION
**Pull request by guohaoyu110**
## Related Issue(s)
Close: https://github.com/kyverno/policies/issues/578

## Description
GitHub Actions step in the [existing ci.yml workflow](https://github.com/kyverno/policies/blob/main/.github/workflows/ci.yml#L21) to validate some of the required items in the policy [Contribution guide](https://github.com/kyverno/policies#contribution). This will fire upon any new PR to kyverno/policies. See examples such as [this one](https://github.com/kyverno/kyverno/blob/main/.github/workflows/cli.yaml#L29-L37).
Policy is located in a folder with no other characters than dashes
Policy is named the same as the folder
The artifacthub-pkg.yml file is found in the same folder as the policy

## Checklist

- [x] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [x] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.
- [x] I have added the artifacthub-pkg.yml file and have verified it is complete and correct.

### CODE REVIEW

**Review by chipzoller — COMMENTED** [review]
This does not appear to have the check for underscores.

**Comment by chipzoller** [comment]
Also see failures. This logic should only work on folders in this repo (not the working directory of the runner) which do not begin with a dot.

**Comment by chipzoller** [comment]
Update?

**Comment by guohaoyu110** [comment]
please review, i think i complete all the requirements. @chipzoller

**Comment by guohaoyu110** [comment]
do i need to also include this check in this?

```
# Check if the directory name only contains alphanumeric characters and dashes
  if [[ ! $dir_name =~ ^[a-zA-Z0-9-]+$ ]]; then
    echo "Directory $dir_name contains invalid characters. Only alphanumeric characters and dashes are allowed."
    exit 1
  fi

```

**Comment by guohaoyu110** [comment]
> Yes, all items listed in the linked issue must be addressed for this PR to fully close the issue.

sure, i add this check to the ci.yml now.

**Comment by guohaoyu110** [comment]
@chipzoller please review.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -35,3 +35,45 @@ jobs:
...
+            # Check if the directory contains any other directories
+            if find "$policy_dir" -mindepth 1 -type d | read; then
+              # If it does, skip the filename validation
+              continue
+            fi
+
+            # Check if a .yaml file with the same name as the directory exists in the directory
+            if [[ ! -f "$policy_dir/$dir_name.yaml" ]]; then
+              echo "No file named $dir_name.yaml found in directory $policy_dir"
+              exit 1
```
Comment (chipzoller) [inline .github/workflows/ci.yml:69]: The file extensions `.yaml` and `.yml` are valid and should be accepted.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -35,3 +35,45 @@ jobs:
...
+            fi
+
+            # Validate that artifacthub-pkg.yml file is found in the same folder as the policy
+            if [[ ! -f "$policy_dir/artifacthub-pkg.yml" ]]; then
+              echo "artifacthub-pkg.yml file is not found in the same folder as the policy in directory $policy_dir"
+              exit 1
+            fi
```
Comment (chipzoller) [inline .github/workflows/ci.yml:76]: The file name `artifacthub-pkg.yml` and `artifacthub-pkg.yaml` is valid and should be accepted.

**Review by chipzoller — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (guohaoyu110) [reply .github/workflows/ci.yml]: updated.

**Comment by chipzoller** [comment]
Please resolve conversations as changes are pushed which address them.

**Comment by chipzoller** [comment]
Changes look good. I will merge this. Once this is merged, please complete three positive tests which cover the scenarios reflected in the parent issue to ensure the changes are effective. Each test should be in a separate PR. The expectation is the Action for each PR fails.

1. Create a new folder with a name which includes dashes.
2. Create a folder and policy where the policy name does not correspond to the folder name.
3. Create a new folder containing a policy which does correspond to the folder name but does not have an artifacthub-pkg.yml file.

**Review by chipzoller — APPROVED** [review]

**Comment by guohaoyu110** [comment]
> 

ok understand.

**Pull request merged**